tool/tsh/kubectl.go

```diff
@@ -330,3 +388,166 @@ func getKubeClusterName(args []string, teleportClusterName string) (string, erro
+// kubeconfig location are returned.
+func makeAndStartKubeLocalProxy(cf *CLIConf, config *clientcmdapi.Config, clusters kubeconfig.LocalProxyClusters) (func(), string, error) {
+	tc, err := makeClient(cf)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+
+	localProxy, err := makeKubeLocalProxy(cf, tc, clusters, config, cf.LocalProxyPort)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+
+	go localProxy.Start(cf.Context)
+
+	closeFn := func() {
+		localProxy.Close()
+	}
+	return closeFn, localProxy.KubeConfigPath(), nil
+}
+
+// shouldUseKubeLocalProxy checks if a local proxy is required for kube
+// access.
+//
+// The local proxy is required when all of these conditions are met:
+// - profile is loadable
+// - kube access is enabled
+// - ALPN connection upgrade is required (e.g. Proxy behind ALB)
+// - not `kubectl config` commands
+// - original/default kubeconfig is loadable
+// - Selected cluster is a Teleport cluster that uses KubeClusterAddr
+func shouldUseKubeLocalProxy(cf *CLIConf, kubectlArgs []string) (*clientcmdapi.Config, kubeconfig.LocalProxyClusters, bool) {
+	// When failed to load profile, assume this CLI command is not running
+	// against Teleport clusters.
+	profile, err := cf.GetProfile()
+	if err != nil {
+		return nil, nil, false
+	}
+
+	// Only use local proxy when Kube is enabled and ALPN connection upgrade is
+	// required.
+	if !profile.TLSRoutingConnUpgradeRequired || profile.KubeProxyAddr == "" {
+		return nil, nil, false
+	}
+
+	// Skip "kubectl config" commands.
+	var kubeconfigLocation, selectedContext string
+	if len(kubectlArgs) > 0 {
+		kubectlCommand := makeKubectlCobraCommand()
+		if isKubectlConfigCommand(kubectlCommand, kubectlArgs) {
+			return nil, nil, false
+		}
+
+		kubeconfigLocation, selectedContext = extractKubeConfigAndContextFromCommand(kubectlCommand, kubectlArgs)
+	}
+
+	// Nothing to do if cannot load original kubeconfig.
+	defaultConfig, err := kubeconfig.Load(kubeconfigLocation)
+	if err != nil {
+		return nil, nil, false
+	}
+
+	// Prepare Teleport kube cluster based on selected context.
+	kubeCluster, found := kubeconfig.FindTeleportClusterForLocalProxy(defaultConfig, kubeClusterAddrFromProfile(profile), selectedContext)
+	if !found {
+		return nil, nil, false
+	}
+	return defaultConfig, kubeconfig.LocalProxyClusters{kubeCluster}, true
+
+}
+
+func isKubectlConfigCommand(kubectlCommand *cobra.Command, args []string) bool {
+	if len(args) < 2 || args[0] != "kubectl" {
+		return false
+	}
+
+	find, _, _ := kubectlCommand.Find(args[1:])
+	for ; find != nil; find = find.Parent() {
+		if find.Name() == "config" {
+			return true
+		}
+	}
+	return false
+}
+
+func kubeClusterAddrFromProfile(profile *profile.Profile) string {
+	partialClientConfig := client.Config{
+		WebProxyAddr:  profile.WebProxyAddr,
+		KubeProxyAddr: profile.KubeProxyAddr,
+	}
+	return partialClientConfig.KubeClusterAddr()
+}
+
+func hasKubeconfigFlagInArgs(args []string) bool {
```

Comment:
you can use  `extractKubeConfigAndContext` that already parses the kubeconfig flag from the args using cobra.

kubeConfig, _:=extractKubeConfigAndContext(args)
return kubeConfig!=""